rotkehlchen/chain/ethereum/modules/lido/decoder.py

```diff
@@ -0,0 +1,526 @@
+            log.error(
+                'At lido steth submit decoding of tx '
+                'not found the related stETH token generation in the same transaction',
+            )
+            return DEFAULT_DECODING_OUTPUT
+
+        # Searching for the already decoded event,
+        # containing the ETH transfer of the submit transaction
+        paired_event = None
+        for event in context.decoded_events:
+            if (
+                event.address == steth_evm_address and
+                event.asset == A_ETH and
+                event.balance.amount == collateral_amount and
+                event.event_type == HistoryEventType.SPEND and
+                event.location_label == sender_address
+            ):
+                event.event_type = HistoryEventType.DEPOSIT
+                event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                event.notes = f'Submit {collateral_amount} {A_ETH.symbol_or_name()} to Lido for receiving {A_STETH.symbol_or_name()} in exchange'  # noqa: E501
+                event.counterparty = CPT_LIDO_ETH
+                #  preparing next action to be processed when erc20 transfer will be decoded by rotki  # noqa: E501
+                #  needed because submit levent is emitted prior of erc20 transfer, so it is not decoded yet  # noqa: E501
+                # TODO: to be confirmed with ROTKI team if it is not possible to have the erc20 event available before this decoder is called  # noqa: E501
+                paired_event = event
+                action_from_event_type = HistoryEventType.RECEIVE
+                action_to_event_subtype = HistoryEventSubType.RECEIVE_WRAPPED
+                action_to_notes = f'Receive {{amount}} {A_STETH.symbol_or_name()} in exchange of the deposited {A_ETH.symbol_or_name()}'  # {amount} to be replaced in post decoding  # noqa: E501
+                break
+
+        action_items = []  # also create an action item for the reception of the stETH tokens
+        if paired_event is not None and action_from_event_type is not None:
+            action_items.append(ActionItem(
+                action='transform',
+                from_event_type=action_from_event_type,
+                from_event_subtype=HistoryEventSubType.NONE,
+                asset=A_STETH,
+                amount=steth_minted_tokens,
+                to_event_subtype=action_to_event_subtype,
+                to_notes=action_to_notes,
+                to_counterparty=CPT_LIDO_ETH,
+                paired_event_data=(paired_event, True),
+                extra_data={'staked_eth': str(collateral_amount)},
+            ))
+        else:  # did not break/find anything
+            log.error(
+                'At lido steth submit decoding of tx '
+                'not found the decoded event of the ETH transfer',
+            )
+            return DEFAULT_DECODING_OUTPUT
+
+        return DecodingOutput(action_items=action_items, matched_counterparty=CPT_LIDO_ETH)
```

Comment:
So for this rule I don't understand what you are trying to do. Why bother with this issue at all? 

The actual transfer from 0x0000 to the sender maps **exactly** to the amount of the internal ETH transfer.

So you really need to do nothing else other than intercept that transfer of stETH and check if the sender is the zero address. If it is then get the corresponding already decoded event of ETH transfer and you are done. All in a simple decoder. 

![2024-05-19_00-05_1](https://github.com/rotki/rotki/assets/1658405/95954388-f57e-487b-8f62-7f66a4a6ca1f)
![2024-05-19_00-05](https://github.com/rotki/rotki/assets/1658405/328dfebb-09d8-4e25-bbb5-c4b81c1fcce4)

Reply:
Unfortunately the amount is not the same, there is 1 wei difference in your example, as in the one I added in the test cases. Please see my reply to the previous comment, we can continue there.